src/screens/ImportAccounts/DisplayResultItem.js

```diff
@@ -1,49 +1,193 @@
-  mode: *,
-  checked: boolean,
-  importing: boolean,
-  onSwitch: (boolean, Account) => void,
-}> {
+class DisplayResultItem extends Component<
+  {
+    account: Account,
+    mode: *,
+    checked: boolean,
+    importing: boolean,
+    onSwitch: (boolean, Account) => void,
+    onAccountNameChange: (string, Account) => void,
+    index?: number,
+    navigation: NavigationScreenProp<*>,
+    signalOtherRows: (?number) => void,
+    openIndex?: number,
+  },
+  { animEnded: boolean },
+> {
+  state = {
+    animEnded: false,
+  };
+
+  panResponder: PanResponder;
+  swipeableRow: Swipeable;
+
   onSwitch = () => {
     track(this.props.checked ? "AccountSwitchOff" : "AccountSwitchOn");
     this.props.onSwitch(!this.props.checked, this.props.account);
   };
 
+  renderLeftActions = (progress, dragX) => {
+    const translateX = dragX.interpolate({
+      inputRange: [0, 1000],
+      outputRange: [-112, 888],
+    });
+
+    return (
+      <Animated.View
+        style={[
+          styles.leftAction,
+          { transform: [{ translateX }] },
+          { marginLeft: 12 },
+        ]}
+      >
+        <Button
+          event="HardResetModalAction"
+          type="primary"
+          title="Edit Name"
+          onPress={this.editAccountName}
+          containerStyle={styles.buttonContainer}
+        />
+      </Animated.View>
+    );
+  };
+
+  updateRef = ref => {
+    if (ref) this.swipeableRow = ref;
+  };
+
+  editAccountName = () => {
+    const { account, navigation, signalOtherRows } = this.props;
+    signalOtherRows(-1);
+
+    navigation.navigate("EditAccountName", {
+      onAccountNameChange: this.onAccountNameChange.bind(this),
+      account,
+      accountId: -1,
+    });
+  };
+
+  onAccountNameChange = name => {
+    this.props.onAccountNameChange(name, this.props.account);
+  };
+
+  componentDidUpdate() {
+    const { openIndex, index } = this.props;
+
+    // If the open index doesn't match us, some other row is open
+    if (openIndex !== index && this.swipeableRow) {
+      this.swipeableRow.close();
+    }
+  }
+
+  componentDidMount() {
+    // const { index } = this.props;
+    //
+    // if (index === 0) {
+    //   // No easy way to emulate touches without an extra library
+    //   // Opens/closes 1st row with modified styles, looks ok ¯\_(ツ)_/¯
+    //   setTimeout(() => this.swipeableRow.openLeft(), 200);
+    //   setTimeout(() => this.swipeableRow.close(), 1000);
+    //   setTimeout(() => this.setState({ animEnded: true }), 1400);
+    // }
+  }
+
+  constructor(props) {
+    super(props);
+    this.panResponder = PanResponder.create({
+      // Ask to be the responder:
+      onStartShouldSetPanResponder: () => true,
+      onStartShouldSetPanResponderCapture: () => false,
+      onMoveShouldSetPanResponder: () => false,
+      onMoveShouldSetPanResponderCapture: () => false,
+
+      onPanResponderGrant: () => {
+        const { index, signalOtherRows } = this.props;
+        signalOtherRows(index);
```

Comment:
Suggestion: instead of `signalOtherRows` and `openIndex`, I would suggest to have a global Subject that you push the index into it.

```js
// global in a file:
addAccountOpeningRows = new Subject()

// when you start interacting with a row:
addAccountOpeningRows.next(index)

// a row that want to listen if another row have an interaction:
addAccountOpeningRows.subscribe(openedIndex => { if (openedIndex !== index) { thing.close() } })
```

this could simplify a bit the parent<>child architecture and avoid some re-rendering.